staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -75,6 +77,24 @@ func newMutatingDispatcher(p *Plugin) func(cm *webhookutil.ClientManager) generi
 	}
 }
 
+var _ generic.VersionedAttributeAccessor = &versionedAttributeAccessor{}
+
+type versionedAttributeAccessor struct {
+	versionedAttr *admission.VersionedAttributes
+}
+
+func (v *versionedAttributeAccessor) VersionedAttribute(attr admission.Attributes, o admission.ObjectInterfaces, gvk schema.GroupVersionKind) (*admission.VersionedAttributes, error) {
+	if v.versionedAttr != nil {
+		return v.versionedAttr, nil
```

Comment:
This is ignoring the input kind, which means we're not necessarily returning the attributes in the right version.

An object which is served at two GVKs, is matched by two webhooks using `matchPolicy: Equivalent`, and which requires conversion for the second webhook would not have been converted correctly for the conditions to evaluate against.

Make sure there's a test that exercises this scenario and fails with the current code before fixing this.

I also added logging here and this function was never called when running unit tests... are we exercising conditions in unit tests?

Reply:
So the code does still convert the versionedAttr after returning it for mutating webhooks 

```
		// Subsequent webhook, convert existing versioned attributes to this webhook's version
		var err error
		versionedAttr, err = v.VersionedAttribute(attr, o, invocation.Kind)
		if err != nil {
			return apierrors.NewInternalError(err)
		}
		if err = admission.ConvertVersionedAttributes(versionedAttr, invocation.Kind, o); err != nil {
			return apierrors.NewInternalError(err)
		}
```

Reply:
so that is a little sketchy since we're passing the pointer and converting it... but I think that would get converted? IIUC (which I may not)

Reply:
Strange, I can hit both return values with a breakpoint when running plugin_test.go. It isn't exercised by dispatch_test.go though

Reply:
So this conversion is still happening in this code https://github.com/ivelichkovich/kubernetes/blob/463d9148cf9c8c6b3a5c25c675b259c17092786c/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go#L149-L156

```
		// Subsequent webhook, convert existing versioned attributes to this webhook's version
		var err error
		versionedAttr, err = v.VersionedAttribute(attr, o, invocation.Kind)
		if err != nil {
			return apierrors.NewInternalError(err)
		}
		if err = admission.ConvertVersionedAttributes(versionedAttr, invocation.Kind, o); err != nil {
			return apierrors.NewInternalError(err)
		}
```
It seems a little sketchy that `admission.ConvertVersionedAttributes(versionedAttr, invocation.Kind, o)` is converting the nested pointer in the `versionedAttributeAccessor` so I can clean that up.

I have implemented an integration test though that checks for the versionedAttr conversion

Reply:
test at `test/integration/apiserver/admissionwebhook/mutating_webhook_gvk_conversion_test.go`